interfaces/apparmor/backend_test.go

```diff
@@ -1844,6 +1847,127 @@ func (s *backendSuite) TestSetupSnapConfineGeneratedPolicyWithOverlayAndReExec(c
+	c.Assert(ioutil.WriteFile(profilePath, []byte(""), 0644), IsNil)
+
+	// Setup generated policy for snap-confine.
+	err := (&apparmor.Backend{}).Initialize(nil)
+	c.Assert(err, IsNil)
+
+	// capability bpf is supported by the parser, so an extra policy file
+	// for snap-confine
+	files, err := ioutil.ReadDir(dirs.SnapConfineAppArmorDir)
+	c.Assert(err, IsNil)
+	c.Assert(files, HasLen, 1)
+	c.Assert(files[0].Name(), Equals, "bpf")
+	c.Assert(files[0].Mode(), Equals, os.FileMode(0644))
+	c.Assert(files[0].IsDir(), Equals, false)
+
+	c.Assert(filepath.Join(dirs.SnapConfineAppArmorDir, files[0].Name()),
+		testutil.FileContains, "capability bpf,")
+
+	if reexec {
+		// The distribution policy was not reloaded because snap-confine executes
+		// from core snap. This is handled separately by per-profile Setup.
+		c.Assert(cmd.Calls(), HasLen, 0)
+	} else {
+		c.Assert(cmd.Calls(), DeepEquals, [][]string{{
+			"apparmor_parser", "--replace",
+			"--write-cache",
+			"-O", "no-expr-simplify",
+			"--cache-loc=" + apparmor_sandbox.SystemCacheDir,
+			"--skip-read-cache",
+			"--quiet",
+			profilePath,
+		}})
+	}
+}
+
+func (s *backendSuite) TestSetupSnapConfineGeneratedPolicyWithBPFCapabilityReexec(c *C) {
+	const reexecd = true
+	s.testSetupSnapConfineGeneratedPolicyWithBPFCapability(c, reexecd)
+}
+
+func (s *backendSuite) TestSetupSnapConfineGeneratedPolicyWithBPFCapabilityNoReexec(c *C) {
+	const reexecd = false
+	s.testSetupSnapConfineGeneratedPolicyWithBPFCapability(c, reexecd)
+}
+
+// snap-confine policy when overlay is used and snapd has re-executed.
+func (s *backendSuite) TestSetupSnapConfineGeneratedPolicyWithBPFProbeError(c *C) {
+	log, restore := logger.MockLogger()
+	defer restore()
+	restore = apparmor.MockIsRootWritableOverlay(func() (string, error) { return "", nil })
+	defer restore()
+	restore = apparmor.MockIsHomeUsingNFS(func() (bool, error) { return false, nil })
+	defer restore()
+	// Probing for apparmor_parser features failed
+	apparmor_sandbox.MockFeatures(nil, nil, nil, fmt.Errorf("mock probe error"))
+
+	// Hijack interaction with apparmor_parser
+	cmd := testutil.MockCommand(c, "apparmor_parser", "")
+	defer cmd.Restore()
+
+	fakeExe := filepath.Join(s.RootDir, "fake-proc-self-exe")
+	restore = apparmor.MockProcSelfExe(fakeExe)
+	defer restore()
+	// Pretend snapd is executing from the native package
+	err := os.Symlink("/usr/lib/snapd/snapd", fakeExe)
+	c.Assert(err, IsNil)
+
+	profilePath := filepath.Join(apparmor_sandbox.ConfDir, "usr.lib.snapd.snap-confine")
+	// Create the directory where system apparmor profiles are stored and write
+	// the system apparmor profile of snap-confine.
+	c.Assert(os.MkdirAll(apparmor_sandbox.ConfDir, 0755), IsNil)
+	c.Assert(ioutil.WriteFile(profilePath, []byte(""), 0644), IsNil)
+
+	// Setup generated policy for snap-confine.
+	err = (&apparmor.Backend{}).Initialize(nil)
+	c.Assert(err, IsNil)
+
+	// capability bpf is supported by the parser, so an extra policy file
```

Comment:
I think this comment should read:

```suggestion
	// capability bpf is not supported by the parser, so no extra policy file
```